docs/on_simulated_device.txt

```diff
@@ -0,0 +1,86 @@
+                                                 +---------+
```

Comment:
For all of these diagrams, could you render SVGs from your mermaid code and use those instead?

Reply:
A potential downside to using the rendered SVG is that it will be stripped when published to docs.rs. 

I think the public documentation for ISO mDL (https://www.sprucekit.dev/#coming-soon) is still WIP, but it may make sense to include the rendered SVG of this diagram to those documentation pages, and link the rust docs to the flow.

Reply:
@cobward @Ryanmtate if mermaid is rendered in docs.io but SVG not, then maybe we could stick to mermaid?
Alternative would be to use ASCII diagrams like we do in lib.rs https://github.com/spruceid/isomdl/blob/main/src/lib.rs#L17

Reply:
@radumarias The ASCII diagrams are very helpful. I don't think SVGs will appear in published rust docs to docs.rs. So we should have a SVG diagram for less technical individuals. I can migrate your diagrams to mermaid spec, unless you already have one started.

I am using your documentation comments and diagrams to work on the GitBook entry for ISO mDL. I pulled in your working branch and I am making some suggestions to the Rust docs that I'd like to upstream to this branch, but it is still WIP. I'll try to get something pushed before EOD.

Reply:
thank you, looking forward to that. what do you mean by
> I can migrate your diagrams to mermaid spec

The ASCII diagrams are present already in mermaid format in the readme file. actually I created those first, then the ASCII ones, which I did almost manually as I didn't find a good converter from mermaid or umlplant, if you know one would appreciate it

Reply:
That's great to have the mermaid text. Thank you for pointing it out!

Mermaid has first-class integration in GitBook, so I will re-use these diagrams, and they can serve as the source of truth for the downstream GitBook user guides.

I don't know of intermediate representation language that will go from SVG to ASCII that is available today. I know d2 has it on their road map (https://d2lang.com/tour/future/#plan-2) but I don't think it is implemented today.

Reply:
> I don't know of intermediate representation language that will go from SVG to ASCII that is available today. I know d2 has it on their road map (https://d2lang.com/tour/future/#plan-2) but I don't think it is implemented today.

Was asking more of how to convert mermaid syntax to ASCII diagrams.

Reply:
It still appears to be an open issue for Mermaid: https://github.com/mermaidjs/mermaid-live-editor/issues/36

Plantuml appears to provide an ASCII converter, which may be quicker to speed up creating ASCII diagrams: https://plantuml.com/ascii-art

Reply:
thank you I guess I tried that also but getting Null pointer exceptions, which is the same on their site and also in draw.io, I think they all use the same service which uses that tool
But will give it one more try